hardware_interface/CMakeLists.txt

```diff
@@ -52,6 +54,7 @@ if(BUILD_TESTING)
 
   ament_add_gtest(test_macros test/test_macros.cpp)
   target_include_directories(test_macros PRIVATE include)
+  ament_target_dependencies(test_macros "rcpputils")
```

Comment:
this is only relevant when building rcpputils from source in the same workspace. Unless there's any other reason I'd opt for not cluttering the cmake more

Reply:
What? I don’t understand this.
The test is relying on the ‘macros.hpp’ which itself depends on ‘rcpputils’. There is no target interface for this, therefore the dependency has to be attached to the test as well as exported for other packages to be usable.

Reply:
ah sorry, I misread it as `add_dependencies`, forget it